test/llvm/dbg-records/dbg-proc-declare.skipif

```diff
@@ -0,0 +1,3 @@
+# under newer LLVM versions, `x` is entirely optimized away
+# and no `call void @llvm.dbg.declare` appears in the IR.
+COMPOPTS <= --fast
```

Comment:
Do we have an issue about improving the debug information we get through LLVM? If not we should make one. Then, please note these degredations (this  and dbg-intrinsics being skipped) in the issue.

Reply:
These are not new degradations. LLVM 19 changed how debug information is printed and represented, see https://releases.llvm.org/19.1.0/docs/ReleaseNotes.html#changes-to-the-llvm-ir and https://llvm.org/docs/RemoveDIsDebugInfo.html.

This is a copy of the old `dbg-intrinsic` test that is tested only with LLVM 19. The `dbg-intrinsic` tests are tested with prior versions of LLVM. The only difference is I changed the `FileCheck` text to match the new version

Reply:
Sounds good, but I don't see any sign of a 2nd version of this test or the dbg-intrinsic test. Where are the LLVM 19 and LLVM < 19 versions?

Reply:
The `test/llvm/dbg-records` folder is the LLVM 19+ version, the `test/llvm/dbg-intrinsics` folder is the LLVM < 19 version. The `dbg-intrinsics` are unchanged (just a new SKIPIF) so they don't show up in the diff

Reply:
ooh I had no idea these were related. I guess `-records` or `-intrinsics` refers to how the debug information is represented in LLVM. If you want to avoid this confusion in the future, you could add READMEs to these, or rename them. Anyway, thanks for explaining.

Reply:
I just pushed some readmes to avoid future confusion